server/src/main/java/org/elasticsearch/action/admin/indices/retention/TransportSyncRetentionLeasesAction.java

```diff
@@ -78,7 +79,16 @@ protected SyncRetentionLeasesRequest readRequestFrom(StreamInput in) throws IOEx
     @Override
     protected void shardOperation(SyncRetentionLeasesRequest request, ShardRouting shardRouting, ActionListener<ReplicationResponse> listener) throws IOException {
         IndexShard indexShard = indicesService.indexServiceSafe(shardRouting.shardId().getIndex()).getShard(shardRouting.shardId().id());
-        indexShard.syncRetentionLeases(true, listener);
+        indexShard.runUnderPrimaryPermit(
+            () -> indexShard.syncRetentionLeases(true, listener),
+            e -> {
+                logger.warn(new ParameterizedMessage("failed to sync retention leases on OPTIMIZE TABLE"), e);
```

Comment:
Would it be much more helpful to mention the shard the retention lease sync failed rather than mention who called this API/action?
I'd prefer not to spread use-cases/callers of API's into the API -> remove any OPTIMIZE mention here. If we still want to log some OPTIMIZE specific message, it should be done on the caller side.

Reply:
Thanks, I will follow up on this

Reply:
https://github.com/crate/crate/pull/17008